src/platform/backends/lxd/lxd_vm_image_vault.cpp

```diff
@@ -207,18 +207,46 @@ mp::VMImage mp::LXDVMImageVault::fetch_image(const FetchType& fetch_type, const
         // Image doesn't exist, so move on
     }
 
-    // TODO: Remove once we do support http & file based images
-    if (query.query_type != Query::Type::Alias)
-        throw std::runtime_error("http and file based images are not supported");
-
-    const auto info = info_for(query);
-    const auto id = info.id;
     VMImage source_image;
+    VMImageInfo info;
+    QString id;
+
+    if (query.query_type == Query::Type::Alias)
+    {
+        info = info_for(query);
+        id = info.id;
+
+        source_image.id = id.toStdString();
+        source_image.original_release = info.release_title.toStdString();
+        source_image.release_date = info.version.toStdString();
+        source_image.aliases = copy_aliases(info.aliases);
+    }
+    else
+    {
+        QUrl image_url(QString::fromStdString(query.release));
+        QDateTime last_modified;
+
+        if (query.query_type == Query::Type::HttpDownload)
+        {
+            // Generate a sha256 hash based on the URL and use that for the id
+            id = QString(QCryptographicHash::hash(query.release.c_str(), QCryptographicHash::Sha256).toHex());
```

Comment:
Will LXD be OK with the id being different than a hash of the image itself?

Reply:
Yes, LXD generates it's own hash when an image is imported into it- we just create this hash for our internal use and is mapped to the LXD hash once we get that back when it imported.

Reply:
Right. I remember now.